webplugin/js/app/km-utils.js

```diff
@@ -324,25 +324,15 @@ KommunicateUtils = {
         }
     },
     getDomainFromUrl: function (hostName) {
-        hostName = hostName || parent.window.location.hostname;
-        var domain = "";
-        if (hostName != null) {
-            var parts = hostName.split('.').reverse();
-
-            if (parts != null && parts.length <= 1) {
-                // cases like  localhost
-                return domain;
-            } else if (parts != null && parts.length <= 3) {
-                // cases:  url with one or no sub domain 
-                domain = "." + parts[1] + '.' + parts[0];
-            } else if (parts != null && parts.length <= 4) {
-
-                if (!parseInt(parts[1]) && !parseInt(parts[0])) {
-                    // check if url is not IP address
-                    domain = "." + parts[2] + "." + parts[1] + "." + parts[0];
-                }
-            }
+        var i = 0;
+        var domain = document.domain;
+        var parts = domain.split('.');
+        var value = 'km_' + (new Date()).getTime();
+        while (i < (parts.length - 1) && document.cookie.indexOf(value + '=' + value) == -1) {
+            domain = parts.slice(-1 - (++i)).join('.');
+            document.cookie = value + "=" + value + ";domain=" + domain + ";";
         }
+        document.cookie = value + "=;expires=Thu, 01 Jan 1970 00:00:01 GMT;domain=" + domain + ";";
```

Comment:
@jithinbalu I don't think we need to do this step.
Tried on jsFiddle, working fine without this also.

Reply:
@ab14bhardwaj yes it will work. 
km_{time_stamp} we are storing into the cookie, so this line is using to clear that data. we are not using that for any purpose.

Reply:
@jithinbalu Can you add comments for the above code like what's happening inside it.
It would be helpful for others in the future. 
I am approving the PR but please add it before merging it.

Reply:
@jithinbalu You can also try this. This is also working and giving proper domain.
It's a small solution. We can use this if it fulfills our use cases.
```
window.location.hostname.split(/[/?#]/)[0];
```
For more information refer below link,
http://scratch99.com/web-development/javascript/how-to-get-the-domain-from-a-url/

Reply:
@ab14bhardwaj comments are added.
The solution you have shared doesn't match with the requirement.

Example: https://docs.kommunicate.io/
Query: window.location.hostname.split(/[/?#]/)[0];
output: docs.kommunicate.io
expected output: kommunicate.io